pkg/executor/join/base_join_probe.go

```diff
@@ -149,6 +149,11 @@ type baseJoinProbe struct {
 	spilledIdx []int
 
 	probeCollision uint64
+
+	memoryUsagePerRowBuffer []int64
```

Comment:
it seems to me this buffer is only used for serializedKey, maybe we can use a more specific name?